src/main/java/io/openliberty/tools/intellij/util/LibertyProjectUtil.java

```diff
@@ -209,19 +201,28 @@ private static boolean isLibertyProject(PsiFile buildFile) {
         return new File(rootDir, "src/main/liberty/config/server.xml").exists();
     }
 
-    private static ShellTerminalWidget getTerminalWidget(ToolWindow terminalWindow, String projectName) {
-        Content[] terminalContents = terminalWindow.getContentManager().getContents();
-        for (int i = 0; i < terminalContents.length; i++) {
-            // TODO use LibertyModule rather than projectName see https://github.com/OpenLiberty/liberty-tools-intellij/issues/143
-            if (terminalContents[i].getTabName().equals(projectName)) {
-                JBTerminalWidget widget = TerminalView.getWidgetByContent(terminalContents[i]);
-                ShellTerminalWidget shellWidget = (ShellTerminalWidget) Objects.requireNonNull(widget);
-                return shellWidget;
+    /**
+     * Get the Terminal widget for the corresponding Liberty module. Will check if the Terminal widget
+     * exists in the Terminal view.
+     *
+     * @param libertyModule
+     * @param terminalView
+     * @return ShellTerminalWidget or null if it does not exist
+     */
+    private static ShellTerminalWidget getTerminalWidget(LibertyModule libertyModule, TerminalView terminalView) {
+        ShellTerminalWidget widget = libertyModule.getShellWidget();
+        // check if widget exists in terminal view
+        if (widget != null) {
+            for (JBTerminalWidget terminalWidget : terminalView.getWidgets()) {
```

Comment:
If someone were to close the associated terminal widget, and we cannot find it anymore, should we call `libertyModule.setShellWidget(null)`?

Reply:
I don't think it'll affect the overall behavior, but rather just updates the state of the LibertyModule instance so we know it no longer has an open Shell Widget

Reply:
I was thinking about this, which is why I was originally setting the widget each time we call `LibertyGeneralAction.getTerminalWidget()`.

However, we are already covered for this case by the method below, which will check if the terminal widget exists in the open terminal view. In the event that the user closes the terminal view themselves, this method will return null so it seems redundant to also set the libertyModule shell widget to null.
https://github.com/OpenLiberty/liberty-tools-intellij/blob/c438c4a2da73c6da5d9a5a32509666cb19aa70ac/src/main/java/io/openliberty/tools/intellij/util/LibertyProjectUtil.java#L204-L223

Reply:
For completeness, I adjusted the above method to call `libertyModule.setShellWidget(null);` if it returns null.